### PR TITLE
[test] introduce mock-grpc module for integration tests

### PR DESCRIPTION
**Pull request by hanfei1991**
use mock-grpc to test communication between master and executors

### CODE REVIEW

File: test/heartbeat_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package test_test
```
Comment (amyangfei) [inline test/heartbeat_test.go:1]: why is this package `test_test`?

File: executor/master-client.go [hunk 2]
```diff
@@ -2,10 +2,12 @@ package executor
 
 import (
 	"context"
-	"errors"
 	"time"
 
 	"github.com/hanfei1991/microcosom/pb"
+	"github.com/hanfei1991/microcosom/pkg/errors"
```
Comment (amyangfei) [inline executor/master-client.go:8]: the import path should be updated

File: test/heartbeat_test.go (hunk 1 above)
Comment (hanfei1991) [reply test/heartbeat_test.go:1]: indeed this is an integration-test. Which package do you think is better?

File: executor/master-client.go (hunk 2 above)
Comment (hanfei1991) [reply executor/master-client.go:8]: shouldn't it be "github.com/hanfei1991/microcosm/pkg/errors" ?

File: test/heartbeat_test.go (hunk 1 above)
Comment (amyangfei) [reply test/heartbeat_test.go:1]: Oh I see, the test is put in a separated package.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**